src/Aspire.Dashboard/Components/Controls/ResourceDetails.razor.cs

```diff
@@ -117,111 +113,101 @@ private IEnumerable<DisplayedEndpoint> GetEndpoints()
-        {
-            KnownResourceTypes.Project => _projectProperties,
-            KnownResourceTypes.Executable => _executableProperties,
-            KnownResourceTypes.Container => _containerProperties,
-            _ => _resourceProperties
-        };
+        PopulateMissingKnownProperties();
 
-        // This is a left outer join for the SQL fans.
-        // Return the resource properties, with an optional known property.
-        // Order properties by the known property order. Unmatched properties are last.
-        var values = Resource.Properties
-            .Where(p => !p.Value.HasNullValue && !(p.Value.KindCase == Value.KindOneofCase.ListValue && p.Value.ListValue.Values.Count == 0))
-            .GroupJoin(
-                resolvedKnownProperties,
-                p => p.Key,
-                k => k.Key,
-                (p, k) => new SummaryValue { Key = p.Key, Value = p.Value, KnownProperty = k.SingleOrDefault(), Tooltip = GetTooltip(p.Value) })
-            .OrderBy(v => v.KnownProperty != null ? resolvedKnownProperties.IndexOf(v.KnownProperty) : int.MaxValue);
-
-        return values;
-    }
+        var vms = Resource.Properties.Values
+            .Where(vm => vm.Value is { HasNullValue: false } and not { KindCase: Value.KindOneofCase.ListValue, ListValue.Values.Count: 0 });
 
-    private static string GetTooltip(Value value)
-    {
-        if (value.HasStringValue)
-        {
-            return value.StringValue;
-        }
-        else
-        {
-            return value.ToString();
-        }
-    }
+        return ordered
+            ? vms.OrderBy(vm => vm.Priority).ThenBy(vm => vm.Name)
+            : vms;
 
-    private static string GetDisplayedValue(BrowserTimeProvider timeProvider, SummaryValue summaryValue)
-    {
-        string value;
-        if (summaryValue.Value.HasStringValue)
-        {
-            value = summaryValue.Value.StringValue;
-        }
-        else
-        {
-            // Complex values such as arrays and objects will be output as JSON.
-            // Consider how complex values are rendered in the future.
-            value = summaryValue.Value.ToString();
-        }
-        if (summaryValue.Key == KnownProperties.Container.Id)
-        {
-            // Container images have a short ID of 12 characters
-            value = value.Substring(0, Math.Min(value.Length, 12));
-        }
-        else
+        void PopulateMissingKnownProperties()
         {
-            // Dates are returned as ISO 8601 text.
-            // Use try parse to check if a value matches ISO 8601 format. If there is a match then convert to a friendly format.
-            if (DateTime.TryParseExact(value, "o", CultureInfo.InvariantCulture, DateTimeStyles.None, out var date))
+            List<KnownProperty>? knownProperties = null;
+
+            // Lazily populate some data on the view models, that wasn't available when they were constructed.
```

Comment:
Longer term, I hope we can avoid patching up these objects by setting these values at construction. For now, this most closely mirrors the code that existed before.

Reply:
This was done in this PR.